source/Examples/SmartCache/Main.cpp

```diff
@@ -6,27 +6,37 @@
 // @File: Main.cpp
 //
 // Purpose:
-//      <description>
+//  The main entrypoint to the SmartCache example.
+//
+//  It's meant as an end-to-end microbenchmark example for the C++ version of the
+//  SmartCache Python Notebook example.
+//
+//  It provides different cache replacement policies and cache sizes as tunables
+//  for an optimizer to tune for different workloads.
 //
 // Notes:
 //      <special-instructions>
 //
 //*********************************************************************
 
+// Include all the common headers for the application
+// (including Mlos.Core and component settings registry code generation output)
+//
 #include "stdafx.h"
 
+// Include platform specific implementations of some Mlos functions.
+// Only needed in one compilation unit for a given target.
+//
 #include "MlosPlatform.Std.inl"
 
 using namespace Mlos::Core;
 using namespace SmartCache;
 
-// Include codegen files.
-//
-#include "GlobalDispatchTable.h"
-
 #include "SmartCacheImpl.h"
 #include "Workloads.h"
 
+// A basic Windows result code handler.
```

Comment:
The curious reader might ask: aren't we running this on linux? ;)

Reply:
Tried to improve that.  Greg has a change coming that actually renames the function to something more meaningful as well.